factories.py

```diff
@@ -0,0 +1,18 @@
+from models import Category
+
+def build_or_update_category(data, category=Category()):
```

Comment:
Two questions:
1. Is `factories` a good name for this package?
2. Can someone come up with a better name for `build_or_update_category`? I just want a method that can take in API params and correctly create or update a category. Not sure if there's a name for that pattern in Python.

Reply:
I'm not so sure about Flask, but in other Python apps I've often seen this sort of logic in a file named `services`, and method names like `update_or_create_category`! It's awfully similar, so I think these work fine :)

Reply:
Thanks for that tip! I'm gonna update that in a follow-up PR.